t/t7519-status-fsmonitor.sh

```diff
@@ -354,4 +354,21 @@ test_expect_success 'discard_index() also discards fsmonitor info' '
 	test_cmp expect actual
 '
 
+# Test staging/unstaging files that appear at the end of the index.  Test
+# files' names begin with 'z' so that they are sorted to the end of the index. 
+test_expect_success 'status succeeds after staging/unstaging ' '
+	test_create_repo fsmonitor-stage-unstage &&
+	(
+		cd fsmonitor-stage-unstage &&
+		test_commit initial &&
+		git update-index --fsmonitor &&
```

Comment:
Interestingly, without this line:

```
git update-index --fsmonitor &&
```

This test would only catch the bug when run with `--run=29`.  If the entire script were run the issue did not repro (event with the new `BUG` statements in this patch).